client/src/error.rs

```diff
@@ -0,0 +1,90 @@
+use std::str;
+
+use serde_json;
+
+#[derive(Debug, Fail)]
+pub enum ClientError {
+    /// What was being attempted when the decode error occured.
+    // https://github.com/rust-lang-nursery/failure/issues/183 prevents
+    // easily deferring rendering until needed.
+    // For now, we render at instantiation as errors are relatively few
+    // we can amortise the costs well.
```

Comment:
I think the right fix/workaround here is to create a new `DecodeError` _struct_, and hand-implement `Display` for that type, avoiding the limitations of the macro-derived display.

This enum variant would then become `DecodeFailed(DecodeError)`

Reply:
Sure, I think that will work

Reply:
Actually no, I think I'm going to pull the 'arwgh too hard' plug here. The processing to do this eagerly is trivial, and error on parsing is a corner case. I think this is good enough - we're into diminishing returns on the effort here, there are much bigger things we can do: the key things - preserve all input, and show it cleanly for operators, are done.

I'll fix the erroneous clones, but thats all for now.

Reply:
sg.